compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -143,7 +143,6 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             Err(e) => e,
         };
 
-        self.set_tainted_by_errors();
```

Comment:
Why is this removed?

Reply:
I actually moved it to the top of `probe_for_return` in the probe module as having it in this location meant that it was being hit when we weren't in an error context during trait bound fulfillment. 

Moving it to the new location meant it only got set when we were actively looking for a method suggestion in an error.

Reply:
Does *keeping* this call to `set_tainted_by_errors` cause any undesired effect?

Reply:
I haven't tried it but can test it out - I think you're right that it can be kept there, I'm fairly sure that both spot I've tried it in are in error reporting pathways anyway. 

I think the bigger issue is that `set_tainted_by_errors` is being called elsewhere before this point due to name resolution being unable to resolve `Thing`

Reply:
Can we try getting this back in to see the effect?

Reply:
Replied in the wrong place - see my comment on the other review but I'm going to push a change tomorrow that I think works better